### DISCUSSION TITLE
Why I set env: { node: false }, but usage of __filename is still allowed?

### DISCUSSION THREAD

**Discussion by lovetingyuan** [eslint/eslint#14942] [Q&A] [answered]
**Tell us about your environment**

* **ESLint Version:** v7.32.0
* **Node Version:** v16.4.0
* **npm Version:** 7.18.1
* **Operating System:** mac os

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
'vue2': {
      env: {
        browser: true,
        es2020: true,
        node: false,
      },
      parserOptions: {
        ecmaVersion: 2020,
        ecmaFeatures: {
          jsx: true,
          legacyDecorators: false
        },
      },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash

```

**What did you expect to happen?**

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

**Steps to reproduce this issue:**

1. 
2. 
3. 

**Are you willing to submit a pull request to fix this bug?**

[a comment by an automated account is not shown]

**Comment by nzakas** [comment]
`__filename` is a valid JavaScript variable so it is allowed in all environments.

**Reply by ljharb** [reply]
I would, however, expect `no-undef` to warn on it in a non-node environment, and not to in a node environment (i haven't checked whether this is the case)

**Comment by lovetingyuan** [comment, marked as answer]
Oh, I found if I use "overrides", the outer "env" option will not work. So I have to set "env" inside override config.

**Reply by ljharb** [reply]
Overrides should merge env values upwards with the root config in the file.

**Reply by lovetingyuan** [reply]
Yes, I thought as you said, but the actual result is not in my case.

**Reply by ljharb** [reply]
That seems like a bug then - could you file an issue?

**Reply by lovetingyuan** [reply]
It is my mistake, the env options work well. There is no problem
